library/dash/src/main/java/com/google/android/exoplayer2/source/dash/DashMediaPeriod.java

```diff
@@ -824,59 +831,96 @@ private static boolean hasEventMessageTrack(List<AdaptationSet> adaptationSets,
     return false;
   }
 
-  private static Format[] getCea608TrackFormats(
+  private static Format[] getClosedCaptionsTrackFormats(
       List<AdaptationSet> adaptationSets, int[] adaptationSetIndices) {
     for (int i : adaptationSetIndices) {
       AdaptationSet adaptationSet = adaptationSets.get(i);
       List<Descriptor> descriptors = adaptationSets.get(i).accessibilityDescriptors;
-      for (int j = 0; j < descriptors.size(); j++) {
-        Descriptor descriptor = descriptors.get(j);
-        if ("urn:scte:dash:cc:cea-608:2015".equals(descriptor.schemeIdUri)) {
-          @Nullable String value = descriptor.value;
-          if (value == null) {
-            // There are embedded CEA-608 tracks, but service information is not declared.
-            return new Format[] {buildCea608TrackFormat(adaptationSet.id)};
-          }
-          String[] services = Util.split(value, ";");
-          Format[] formats = new Format[services.length];
-          for (int k = 0; k < services.length; k++) {
-            Matcher matcher = CEA608_SERVICE_DESCRIPTOR_REGEX.matcher(services[k]);
-            if (!matcher.matches()) {
-              // If we can't parse service information for all services, assume a single track.
-              return new Format[] {buildCea608TrackFormat(adaptationSet.id)};
-            }
-            formats[k] =
-                buildCea608TrackFormat(
-                    adaptationSet.id,
-                    /* language= */ matcher.group(2),
-                    /* accessibilityChannel= */ Integer.parseInt(matcher.group(1)));
-          }
-          return formats;
-        }
+      for (Descriptor descriptor : descriptors) {
+        return parseClosedCaptionsDescriptor(adaptationSet.id, descriptor);
```

Comment:
Might be irrelevant if you can go with my proposal above, but this shouldn't return if `parsedClosedCaptionsDescriptor` returned `null`. Else parsing will fail if there's a descriptor for something other than 608/708 before the descriptor you're looking for.

Reply:
Done